src/iptvsimple/Epg.cpp

```diff
@@ -392,17 +392,17 @@ void Epg::ApplyChannelsLogosFromEPG()
   for (const auto& channel : m_channels.GetChannelsList())
   {
     const ChannelEpg* channelEpg = FindEpgForChannel(channel);
-    if (!channelEpg || channelEpg->GetIcon().empty())
+    if (!channelEpg || channelEpg->GetIconPath().empty())
       continue;
 
     // 1 - prefer logo from playlist
-    if (!channel.GetLogoPath().empty() && Settings::GetInstance().GetEpgLogosMode() == EpgLogosMode::PREFER_M3U)
+    if (!channel.GetIconPath().empty() && Settings::GetInstance().GetEpgLogosMode() == EpgLogosMode::PREFER_M3U)
```

Comment:
If you change code from `GetLogoPath` to `GetIconPath` it might be a good idea to adapt the comment in line 398 accordingly.